components/pluralrules/src/lib.rs

```diff
@@ -0,0 +1,381 @@
+//! [`Plural Type`]: ./enum.PluralRuleType.html
+//! [`Plural Category`]: ./enum.PluralCategory.html
+//! [`Language Plural Rules`]: https://unicode.org/reports/tr35/tr35-numbers.html#Language_Plural_Rules
+//! [`CLDR`]: http://cldr.unicode.org/
+//! [`data`]: ./data/index.html
+pub mod data;
+mod operands;
+pub mod rules;
+
+use icu_locale::LanguageIdentifier;
+pub use operands::PluralOperands;
+
+/// A type of a plural rule which can be associated with the [`PluralRules`] struct.
+///
+/// [`PluralRules`]: ./struct.PluralRules.html
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash)]
+pub enum PluralRuleType {
+    /// Cardinal plural forms express quantities of units such as time, currency or distance,
+    /// used in conjunction with a number expressed in decimal digits (i.e. "2", not "two").
+    ///
+    /// For example, English has two forms for cardinals:
+    ///
+    /// * [`One`]: `1 day`
+    /// * [`Other`]: `0 days`, `2 days`, `10 days`, `0.3 days`
+    ///
+    /// [`One`]: ./enum.PluralCategory.html#variant.One
+    /// [`Other`]: ./enum.PluralCategory.html#variant.Other
+    Cardinal,
+    /// Ordinal plural forms denote the order of items in a set and are always integers.
+    ///
+    /// For example, English has four forms for ordinals:
+    ///
+    /// * [`One`]: `1st floor`, `21st floor`, `101st floor`
+    /// * [`Two`]: `2nd floor`, `22nd floor`, `102nd floor`
+    /// * [`Few`]: `3rd floor`, `23rd floor`, `103rd floor`
+    /// * [`Other`]: `4th floor`, `11th floor`, `96th floor`
+    ///
+    /// [`One`]: ./enum.PluralCategory.html#variant.One
+    /// [`Two`]: ./enum.PluralCategory.html#variant.Two
+    /// [`Few`]: ./enum.PluralCategory.html#variant.Few
+    /// [`Other`]: ./enum.PluralCategory.html#variant.Other
+    Ordinal,
+}
+
+/// The plural categories are used to format messages with numeric placeholders, expressed as decimal numbers.
+///
+/// The fundamental rule for determining plural categories is the existence of minimal pairs: whenever two different
+/// numbers may require different versions of the same message, then the numbers have different plural categories.
+///
+/// All languages supported by `ICU4X` can match any number to one of the categories.
+///
+/// # Examples
+///
+/// ```
+/// use icu_locale::LanguageIdentifier;
+/// use icu_pluralrules::{PluralRules, PluralRuleType, PluralCategory};
+/// use icu_pluralrules::data::provider::DummyDataProvider;
+///
+/// let lang: LanguageIdentifier = "en".parse()
+///     .expect("Failed to parse a language identifier.");
+///
+/// let dp = DummyDataProvider::default();
+///
+/// let pr = PluralRules::try_new(lang, PluralRuleType::Cardinal, &dp)
+///     .expect("Failed to construct a PluralRules struct.");
+///
+/// assert_eq!(pr.select(1_usize), PluralCategory::One);
+/// assert_eq!(pr.select(5_usize), PluralCategory::Other);
+/// ```
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash)]
+pub enum PluralCategory {
```

Comment:
FYI, ICU calls this `StandardPlural`, since there's nothing in UTS 35 preventing someone from using a different name for a custom plural form.

Reply:
How would a custom plural category work? CLDR will never return it, right?

Is it ok for me to use `PluralCategory` here?

Reply:
I'd suggest we go with `Custom(String)` and add it if/when we see a need for it. So far for ECMA402 level coverage we don't I believe.

Reply:
OK, yes, there's a list of the six standard plurals in https://tc39.es/ecma402/#sec-intl.pluralrules.prototype.resolvedoptions.  So let's not act on this issue for now and wait until someone complains.